metrics/resource_view_test.go

```diff
@@ -341,19 +343,32 @@ testComponent_testing_value{project="p1",revision="r2"} 1
 			UnregisterResourceView(gaugeView, resourceCounter)
 
 			records := []metricExtract{}
-			for record := range ocFake.published {
-				for _, m := range record.Metrics {
-					if len(m.Timeseries) > 0 {
-						labels := map[string]string{}
-						if record.Resource != nil {
-							labels = record.Resource.Labels
+		loop:
```

Comment:
Stylistically I'd prefer a lambda with return rather than goto label, but up to you. :)

Reply:
That would look like:

```
for {
  breakErr := error.New("intentional break")
  err := func() error {
    select {
    case record := <-ocFake.published:
      if record == nil {
        return nil
      }
      // ...
      if len(records) >= len(expected) {
        return breakErr
      }
    case <-time.After(5 * time.Second):
      return errors.New("timeout")
    }
  }()
  if err != breakErr {
    t.Fatal("Unable to fetch events", err)
  }
  if err == breakErr {
    break
  }
}

Reply:
I think with the label is better, stylistically.

Reply:
Up to you :-)